dali/util/image.h

```diff
@@ -36,45 +36,88 @@ namespace dali {
+  uint8 *addImage(int imgSize, const uint8 *pRaster = nullptr) {
+    shapes_.push_back({imgSize});
+    return AddRaster(imgSize, pRaster);
+  }
 
-  vector<uint8 *> data_;
-  vector<int> sizes_;
-};
+  uint8 *addImage(int h, int w, int c, const uint8 *pRaster = nullptr) {
+    shapes_.push_back({h, w, c});
+    return AddRaster(h * w * c, pRaster);
+  }
 
-/**
- * Load all images from a list of image names. Assumes names contain
- * full path
- */
-DLL_PUBLIC void LoadImages(const vector<string> &image_names, ImgSetDescr *imgs);
+  inline void addImageName(const string &name)  { img_names_.push_back(name); }
+  const vector<string> &imgNames() const        { return img_names_; }
 
-/**
- * Loads images from a specified image folder. Assumes the folder contains
- * a file 'image_list.txt' that lists all the different images in the
- * folder
- */
-DLL_PUBLIC void LoadImages(const string &image_folder, vector<string> *jpeg_names,
-                           ImgSetDescr *imgs);
+  void copyImage(int idx, void *pRaster) const  { memcpy(pRaster, data(idx), size(idx)); }
 
-/**
- * Loads jpegs from a specified image folder. Assumes the folder contains
- * a file 'image_list.txt' that lists all the different images in the
- * folder
- */
-DLL_PUBLIC void LoadJPEGS(const string &image_folder, vector<string> *jpeg_names,
-                          ImgSetDescr *imgs);
+  inline uint8 *data(int idx) const             { return data_[idx]; }
+  inline int size(int idx) const                { return shape(idx)[0]; }
+  inline const Dims shape(int idx) const        { return shapes_[idx]; }
 
-/**
- * Loads all jpegs from the list of image names. Assumes names contains
- * full path
+ /**
+ * Loads images from a specified image folder. When imgNames is not defined, assumes the folder
+ * contains a file 'image_list.txt' that lists all the different images in the folder
  */
-DLL_PUBLIC void LoadJPEGS(const vector<string> &jpeg_names, ImgSetDescr *imgs);
+  void LoadImages(const string &image_folder, const vector<string> *imgNames = NULL) {
+    if (!imgNames) {
+      const string image_list = image_folder + "/image_list.txt";
+      std::ifstream file(image_list);
+      DALI_ENFORCE(file.is_open());
+
+      string img;
+      while (file >> img) {
+        DALI_ENFORCE(img.size());
+        addImageName(image_folder + "/" + img);
+      }
+    } else {
+      for (auto img : *imgNames) {
+        DALI_ENFORCE(img.size());
+        addImageName(image_folder + img);
+      }
+    }
+
+    LoadImages();
+  }
+
+  void LoadImages(const vector<string> *imgNames = NULL) {
+    if (!imgNames)
+      imgNames = &img_names_;
+
+    for (auto img_name : *imgNames) {
+      std::ifstream img_file(img_name);
+      DALI_ENFORCE(img_file.is_open());
+
+      img_file.seekg(0, std::ios::end);
+      int img_size = static_cast<int>(img_file.tellg());
+      img_file.seekg(0, std::ios::beg);
+
+      auto data = addImage(img_size);
```

Comment:
Could you use "auto *data" for highlighting that type deduced here is a pointer? It improves code readibility.

Reply:
DONE